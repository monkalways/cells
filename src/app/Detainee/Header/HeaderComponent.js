import React from 'react';
import PropTypes from 'prop-types';
import {
  AppBar,
  Grid,
  IconButton,
  Toolbar,
  withStyles,
} from '@material-ui/core';
import CloseIcon from '@material-ui/icons/Close';

import logoIcon from './eps-logo-icon.png';
import HeaderContent from './HeaderContent';

const propTypes = {
  classes: PropTypes.shape({}).isRequired,
  detainee: PropTypes.shape({}),
  isDetaineeProfileLoaded: PropTypes.bool.isRequired,
  onLogout: PropTypes.func.isRequired,
};

const defaultProps = {
  detainee: null,
};

const Header = ({
  classes, detainee, isDetaineeProfileLoaded, onLogout,
}) => (
  <div className={classes.root}>
    <AppBar position="static" className={classes.appBar}>
      <Toolbar>
        <IconButton
          className={classes.menuButton}
          color="inherit"
          aria-label="Menu"
        >
          <img src={logoIcon} alt="EPS Logo" width={70} />
        </IconButton>
        {isDetaineeProfileLoaded ? (
          <HeaderContent
            detainee={detainee}
            isDetaineeProfileLoaded={isDetaineeProfileLoaded}
          />
        ) : (
          <Grid item xs={12} />
        )}
        <IconButton
          className={classes.logoutButton}
          color="inherit"
          aria-label="Menu"
          onClick={onLogout}
        >
          <CloseIcon className={classes.logoutIcon} />
        </IconButton>
      </Toolbar>
    </AppBar>
  </div>
);

Header.propTypes = propTypes;
Header.defaultProps = defaultProps;

export default withStyles((theme) => ({
  root: {
    flexGrow: 1,
  },
  appBar: {
    backgroundColor: '#A8C6FA', // TODO: move color to theme
  },
  menuButton: {
    marginLeft: -12,
    marginRight: 20,
  },
  logoutButton: {
    margin: theme.spacing.unit,
    backgroundColor: '#444480',
    color: '#F10607',
    '&:hover': {
      backgroundColor: '#444480',
    },
  },
  logoutIcon: {
    fontSize: theme.typography.h4.fontSize,
  },
}))(Header);
