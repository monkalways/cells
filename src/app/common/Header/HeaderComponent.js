import React from 'react';
import PropTypes from 'prop-types';
import {
  AppBar, Toolbar, Typography, withStyles,
} from '@material-ui/core';

const propTypes = {
  classes: PropTypes.shape({}).isRequired,
  versions: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    version: PropTypes.string.isRequired,
  })).isRequired,
};

export const HeaderComponent = ({ classes, versions }) => (
  <AppBar position="static">
    <Toolbar>
      <Typography
        id="title"
        variant="h6"
        color="inherit"
        className={classes.heading}
      >
        Cell App - Detainee Management Unit
      </Typography>
      <div>
        {versions
          && versions.map((version) => (
            <Typography
              id="version"
              color="inherit"
              align="right"
              key={version.name}
            >
              {version.name}: {version.version}
            </Typography>
          ))}
      </div>
    </Toolbar>
  </AppBar>
);

HeaderComponent.propTypes = propTypes;

export default withStyles(() => ({
  heading: {
    flex: 1,
  },
}))(HeaderComponent);
