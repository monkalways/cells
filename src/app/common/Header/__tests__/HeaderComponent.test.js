import React from 'react';
import { shallow } from 'enzyme';

import { HeaderComponent } from '../HeaderComponent';

describe('HeaderComponent', () => {
  let classes;
  let versions;

  beforeEach(() => {
    classes = {
      heading: 'heading',
    };
    versions = [
      {
        name: 'Cell App',
        version: '1.0.0',
      },
      {
        name: 'Detainee Management Service',
        version: '1.0.0',
      },
      {
        name: 'EPROS Service',
        version: '1.0.0',
      },
    ];
  });

  const setup = () => shallow(<HeaderComponent classes={classes} versions={versions} />);

  it('should render', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
