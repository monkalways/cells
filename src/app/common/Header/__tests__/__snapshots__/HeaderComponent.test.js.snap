// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderComponent should render 1`] = `
<WithStyles(AppBar)
  position="static"
>
  <WithStyles(Toolbar)>
    <WithStyles(Typography)
      className="heading"
      color="inherit"
      id="title"
      variant="h6"
    >
      Cell App - Detainee Management Unit
    </WithStyles(Typography)>
    <div>
      <WithStyles(Typography)
        align="right"
        color="inherit"
        id="version"
        key="Cell App"
      >
        Cell App
        : 
        1.0.0
      </WithStyles(Typography)>
      <WithStyles(Typography)
        align="right"
        color="inherit"
        id="version"
        key="Detainee Management Service"
      >
        Detainee Management Service
        : 
        1.0.0
      </WithStyles(Typography)>
      <WithStyles(Typography)
        align="right"
        color="inherit"
        id="version"
        key="EPROS Service"
      >
        EPROS Service
        : 
        1.0.0
      </WithStyles(Typography)>
    </div>
  </WithStyles(Toolbar)>
</WithStyles(AppBar)>
`;
